Apps/Playground/Win32/App.cpp

```diff
@@ -398,20 +399,82 @@ LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
         }
         case WM_LBUTTONDOWN:
         {
-            SetCapture(hWnd);
+            if (buttonRefCount++ == 0)
+            {
+                SetCapture(hWnd);
+            }
             if (nativeInput != nullptr)
             {
-                nativeInput->MouseDown(0, GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam));
+                nativeInput->MouseDown(Babylon::Plugins::NativeInput::LEFT_MOUSE_BUTTON_ID, GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam));
             }
             break;
         }
         case WM_LBUTTONUP:
         {
             if (nativeInput != nullptr)
             {
-                nativeInput->MouseUp(0, GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam));
+                nativeInput->MouseUp(Babylon::Plugins::NativeInput::LEFT_MOUSE_BUTTON_ID, GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam));
+            }
+            if (buttonRefCount-- == 1)
```

Comment:
It would be more symmetrical and very slightly faster if you did this. (Ditto everywhere else)
```suggestion
            if (--buttonRefCount == 0)
```

Reply:
Changed